bin/transform-genbank

```diff
@@ -0,0 +1,282 @@
+#!/usr/bin/env python3
+"""
+Parse the GenBank JSON load into a metadata tsv and a FASTA file.
+"""
+import argparse
+from pathlib import Path
+import re
+import pandas as pd
+from typing import List
+from utils.transform import (
+    METADATA_COLUMNS,
+    standardize_dataframe,
+    fill_default_geo_metadata,
+    write_fasta_file,
+)
+
+assert 'sequence' not in METADATA_COLUMNS, "Sequences should not appear in metadata!"
+
+def preprocess(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Rename and standardize column types and drop records where the
+    sequence length is less than 15kkb.
+    Returns the modified DataFrame.
+    """
+    mapper = {
+        'collected': 'date',
+        'submitted': 'date_submitted',
+    }
+
+    # Standardize column names, dtypes and drop entries with length less than 15kb
+    return standardize_dataframe(genbank_data, mapper)
+
+
+def standardize_strain_names(genbank_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Attempt to standardize strain names by removing extra prefixes,
+    stripping spaces, converting country code to country name, and correcting
+    known common error patterns.
+    """
+    # Compile list of regex to be used for strain name standardization
+    # Order is important here! Keep the known prefixes first!
+    regex_replacement = [
+        (r'(^SAR[S]{0,1}[-\s]CoV[-]{0,1}2/|^2019[-\s]nCoV[-_\s/]|^BetaCoV/|^nCoV-|^hCoV-19/)',''),
+        (r'(human/)',''),
+        (r'^USA-', 'USA/'),
+        (r'^USACT-', 'USA/CT-'),
+        (r'^USAWA-', 'USA/WA-'),
+        (r'^HKG.', 'HongKong/'),
+    ]
+    # Add country codes to regex_replacement
+    country_codes = pd.read_csv(base / 'source-data/country-codes.tsv', header=None, sep='\t', comment="#")
+    country_codes.apply(lambda c: regex_replacement.extend([
+        (rf'^{c[0]}/', f'{c[1]}/'),
+        (rf'/{c[0]}/', f'/{c[1]}/')]), axis=1)
```

Comment:
To be safe, it'd be good to call `re.escape` on the values used to construct these regexes.

If I'm understanding correctly, I think the two patterns appended to `regex_replacement` could also be reduced to `(rf'(^|/){c[0]}/', rf'\1{c[1]}/')`. I wonder if this makes the intent more clear or less clear.

Reply:
Yes, it can be reduced to `(rf'(^|/){c[0]}/', rf'\1{c[1]}/')` (although I had to look up what `\1` was doing here). Regex also always looks like gibberish to me so not sure which way is more clear here.

Reply:
I'd say stick with what you had then. :-)